config/routes.rb

```diff
@@ -36,4 +36,15 @@
   get "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinary_offices#veterinarians"
   get "/veterinary_offices/:veterinary_office_id/veterinarians/new", to: "veterinarians#new"
   post "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinarians#create"
+
+  post "/applications", to: "applications#create"
+  get "/applications/new", to: "applications#new"
+  get "/applications/:id", to: "applications#show", as: 'show_application'
+  post "/applications/:id", to: "pet_applications#create", as: 'create_pet_application'
```

Comment:
When we are creating an association between 2 existing resources like we are here, a more RESTful option for the path would be:
POST to "/applications/:application_id/pets/:pet_id" or "application_pets"
Your controller action is very RESTful as you are creating the record in a `create` action in a joins controller, but the path could be more RESTful